### PR TITLE
Changes the code To return the list in dynamic_programming/subset_generation.py

### PR DESCRIPTION
**Pull request by AasheeshLikePanner**
### Describe your change:
Changes the code to return the tuples. So, we can add doctests in it after it. Change the whole code now code using dynamic programming as it header says

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".
#10189

### CODE REVIEW

File: dynamic_programming/subset_generation.py [hunk 1]
```diff
@@ -1,44 +1,25 @@
...
-
-def print_combination(arr, n, r):
-    # A temporary array to store all combination one by one
-    data = [0] * r
-    # Print all combination using temporary array 'data[]'
-    combination_util(arr, n, r, 0, data, 0)
+# return all subset combinations of n element in given set of r element.
+
+
+def subset_combinations_dp(elements, n):
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:4]: No type hints, No docstring, no doctests, `n` is not a self-documenting variable name.

Why ask me to review this pull request when we have 200+ open pull requests that are more in alignment with CONTRIBUTING.md?

File: dynamic_programming/subset_generation.py [hunk 2]
```diff
@@ -1,44 +1,25 @@
...
 
 if __name__ == "__main__":
-    # Driver code to check the function above
-    arr = [10, 20, 30, 40, 50]
-    print_combination(arr, len(arr), 3)
-    # This code is contributed by Ambuj sahu
+    elements = [10, 20, 30, 40]
+    n = 2
+    result = subset_combinations_dp(elements, n)
+    print(result)
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:25]:
```suggestion
    print(f"{subset_combinations_dp(elements=[10, 20, 30, 40], n=2) = }")
```

File: dynamic_programming/subset_generation.py (hunk 1 above)
Comment (AasheeshLikePanner) [reply dynamic_programming/subset_generation.py:4]: Sorry for this mistake, but you previously said to do one thing at  a time.

Comment (cclauss) [reply dynamic_programming/subset_generation.py:4]: I did?

File: dynamic_programming/subset_generation.py [hunk 3]
```diff
@@ -1,7 +1,33 @@
 # return all subset combinations of n element in given set of r element.
 
 
-def subset_combinations_dp(elements, n):
+def subset_combinations_dp(elements: list, n: int) -> list:
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:4]:
```suggestion
def subset_combinations_dp(elements: list[int], n: int) -> list:
```

File: dynamic_programming/subset_generation.py (hunk 1 above)
Comment (AasheeshLikePanner) [reply dynamic_programming/subset_generation.py]: yes

File: dynamic_programming/subset_generation.py [hunk 4]
```diff
@@ -1,44 +1,52 @@
...
+
+    return combinations
 
 
 if __name__ == "__main__":
-    # Driver code to check the function above
-    arr = [10, 20, 30, 40, 50]
-    print_combination(arr, len(arr), 3)
-    # This code is contributed by Ambuj sahu
+    print(f"{subset_combinations_dp(elements=[10, 20, 30, 40], n=2) = }")
```
Comment (cclauss) [inline dynamic_programming/subset_generation.py:48]: `subset_combinations_dp()` should work the same way as `itertools.combinations()`.  It should present the same sort order and raise the same exceptions.
```suggestion
    from itertools import combinations
    for items, n in (
        ([10, 20, 30, 40], 2), ([1, 2, 3], 1), ([1, 2, 3], 3), ([42], 1),
        ([6, 7, 8, 9], 4), ([10, 20, 30, 40, 50], 1), ([1, 2, 3, 4], 2),
        ([1, 'apple', 3.14], 2), (['single'], 0), ([], 9)
    ):
        actual = subset_combinations_dp(items, n)
        expected = set(combinations(items, n))
        assert actual == expected, f"items, n: {actual} != {expected}"


    print(f"{subset_combinations_dp(elements=[10, 20, 30, 40], n=2) = }")
```

Comment (AasheeshLikePanner) [reply dynamic_programming/subset_generation.py:52]: Now code giving the list in sort order

Comment (cclauss) [reply dynamic_programming/subset_generation.py:52]: `itertools.combinations([1, 'apple', 3.14], 2)`

**Review by cclauss — APPROVED** [review]
Thanks for going the extra mile on this one... It is a nice upgrade!

**Comment by AasheeshLikePanner** [comment]
Really challenging and enjoyable. I fix one bug, then two errors comes. It's kind of frustrating, but worth the time.

**Pull request merged**